R/utilities-plots.R

```diff
@@ -77,4 +77,77 @@ autoAxesTicksFromLimits <- function(limits) {
+  plotConfiguration <- eval(parse(text = paste0("tlf::", plotType, "PlotConfiguration$new()")))
+  
+  # Set properties from FontAndSize
+  fonts <- plotProperties$FontAndSize$Fonts
+  plotConfiguration$labels$title$font$size <- reEnv$fontScaleFactor*fonts$TitleSize %||% plotConfiguration$labels$title$font$size
+  plotConfiguration$labels$subtitle$font$size <- reEnv$fontScaleFactor*fonts$DescriptionSize %||% plotConfiguration$labels$subtitle$font$size
+  plotConfiguration$labels$xlabel$font$size <- reEnv$fontScaleFactor*fonts$AxisSize %||% plotConfiguration$labels$xlabel$font$size
+  plotConfiguration$labels$ylabel$font$size <- reEnv$fontScaleFactor*fonts$AxisSize %||% plotConfiguration$labels$ylabel$font$size
+  plotConfiguration$xAxis$font$size <- reEnv$fontScaleFactor*fonts$AxisSize %||% plotConfiguration$xAxis$font$size
+  plotConfiguration$yAxis$font$size <- reEnv$fontScaleFactor*fonts$AxisSize %||% plotConfiguration$yAxis$font$size
+  plotConfiguration$legend$font$size <- reEnv$fontScaleFactor*fonts$LegendSize %||% plotConfiguration$legend$font$size
+  plotConfiguration$background$watermark$font$size <- reEnv$fontScaleFactor*fonts$WatermarkSize %||% plotConfiguration$background$watermark$font$size
+  
+  # If chart size is defined, it is in pixel and updated accordingly
+  # Get conversion factor between pixels and inches, dev.size provides an array c(width, height)
+  unitConversionFactor <- grDevices::dev.size("in") / grDevices::dev.size("px")
+  width <- ifnotnull(
+    plotProperties$FontAndSize$ChartWidth, 
+    plotProperties$FontAndSize$ChartWidth * unitConversionFactor[1],
+    reEnv$defaultPlotFormat$width
+  )
+  height <- ifnotnull(
+    plotProperties$FontAndSize$ChartHeight, 
+    plotProperties$FontAndSize$ChartHeight * unitConversionFactor[2],
+    reEnv$defaultPlotFormat$height
+    )
+  
+  legendScaling <- getLegendScalingFactors(legendPosition)
+  
+  # Get dimensions of exported based on legend position and default/specific plot properties
+  plotConfiguration$export$units <- reEnv$defaultPlotFormat$units
+  plotConfiguration$export$width <- reEnv$fontScaleFactor*legendScaling$width*width
+  plotConfiguration$export$height <- reEnv$fontScaleFactor*legendScaling$height*height
+  return(plotConfiguration)
+}
+
+#' @title getLegendScalingFactors
+#' @description Get factors to scale plot dimensions accounting for the location of the legend.
+#' Initial estimates based on Abdullah's tests
+#' TODO: improve this directly from tlf
+#' @param legendPosition The name of the legend position as defined by `tlf` enum `LegendPositions`
+#' @return A list of scaling values for `width` and `height`
+#' @keywords internal
+getLegendScalingFactors <- function(legendPosition = tlf::LegendPositions$outsideTop){
+  # Legend on the left/right sides: increase width
+  if(isIncluded(legendPosition, c(tlf::LegendPositions$outsideRight, tlf::LegendPositions$outsideLeft))){
+    return(list(width = 1.6, height = 1.2))
+  }

```

Comment:
@pchelle : because this is a scaling that deals specifically with the legend, I think it would be better to scale the width by  4/3 (=1.6/1.2) and keep the height as is (ie scale the height by 1).  This is because the legend only takes up space horizontally.  If we need the 1.2x scaling, it can be introduced as an additional scale factor applied to the whole image elsewhere, in another function.

Reply:
At least for the DDI, I think we will need the additional 1.2x stretch for the plot width and height only.  I don't think it is needed for the font sizes.  So can we introduce, for example

`reEnv$plotScaleFactor <- 2.4`

and then 
```
plotConfiguration$export$width <- reEnv$plotScaleFactor*legendScaling$width*width
plotConfiguration$export$height <- reEnv$plotScaleFactor*legendScaling$height*height
```